githubdelivery/src/main/java/io/dockstore/githubdelivery/GithubDeliveryS3Client.java

```diff
@@ -80,29 +93,79 @@ public static void main(String[] args) throws IOException {
             final GithubDeliveryS3Client githubDeliveryS3Client = new GithubDeliveryS3Client(githubDeliveryConfig.getS3Config().bucket());
-
-            if (DOWNLOAD_COMMAND.equals(jCommander.getParsedCommand())) {
-                System.out.println(githubDeliveryS3Client.getGitHubDeliveryEventByKey(downloadEventCommand.getBucketKey()));
+            if (SUBMIT_EVENT_COMMAND.equals(jCommander.getParsedCommand())) {
+                githubDeliveryS3Client.submitGitHubDeliveryEventsByKey(githubDeliveryConfig, submitEventCommand.getBucketKey());
+            }
+            if (SUBMIT_ALL_COMMAND.equals(jCommander.getParsedCommand())) {
+                githubDeliveryS3Client.submitGitHubDeliveryEventsByDate(githubDeliveryConfig, submitAllEventsCommand.getDate());
             }
         }
 
     }
-    private PushPayload getGitHubDeliveryEventByKey(String key) throws IOException, NoSuchKeyException {
+    private String getObject(String key) throws IOException {
         GetObjectRequest objectRequest = GetObjectRequest
                 .builder()
                 .key(key)
                 .bucket(bucketName)
                 .build();
         ResponseInputStream<GetObjectResponse> object = s3Client.getObject(objectRequest);
-
+        return IOUtils.toString(object, StandardCharsets.UTF_8);
+    }
+    private PushPayload getGitHubPushPayloadByKey(String body, String key) throws IOException, NoSuchKeyException {
         try {
             PushPayload pushPayload;
-            pushPayload = GSON.fromJson(IOUtils.toString(object, StandardCharsets.UTF_8), PushPayload.class);
+            pushPayload = MAPPER.readValue(body, PushPayload.class);
             return pushPayload;
         } catch (JsonSyntaxException e) {
             LOG.error("Could not read github event from key {}", key, e);
             System.exit(1);
         }
         return null;
     }
+    private InstallationRepositoriesPayload getGitHubInstallationRepositoriesPayloadByKey(String body, String key) throws IOException, NoSuchKeyException {
+        try {
+            InstallationRepositoriesPayload installationRepositoriesPayload;
+            installationRepositoriesPayload = MAPPER.readValue(body, InstallationRepositoriesPayload.class);
+            return installationRepositoriesPayload;
+        } catch (JsonSyntaxException e) {
+            LOG.error("Could not read github event from key {}", key, e);
+            System.exit(1);
+        }
+        return null;
+    }
+    private void submitGitHubDeliveryEventsByDate(GithubDeliveryConfig config, String date) {
+
+        ListObjectsV2Request listObjectsV2Request = ListObjectsV2Request
+                .builder()
+                .bucket(bucketName)
+                .prefix(date)
+                .build();
+        ListObjectsV2Iterable listObjectsV2Iterable = s3Client.listObjectsV2Paginator(listObjectsV2Request);
+        for (ListObjectsV2Response response : listObjectsV2Iterable) {
```

Comment:
Do the objects come out of the iterable in ascending date order?  If not, we could sort them so that they are submitted in order.  (might not matter sometimes, but could...)

Reply:
Yes, it seems to come out the iterable in order from earliest to latest